#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
  basicTimeStep 8
}
Viewpoint {
  orientation -0.13928768786466195 0.4769435408768236 0.8678270558269052 0.6492554504133765
  position -1.1468026821766129 -1.113516415748566 0.6547282984408936
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
}
Robot {
  translation 0 0 0.01
  rotation 0 1 0 0
  children [
    Solid {
      translation -0.06 0.01 0.04
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0.309804 0.309804 0.309804
            }
          }
          geometry Box {
            size 0.05 0.02 0.06
          }
        }
      ]
      name "solid(1)"
      boundingObject Box {
        size 0.05 0.02 0.06
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
      }
      device [
        RotationalMotor {
          name "motor_yaw"
        }
      ]
      endPoint Solid {
        translation 0 0 0.02
        rotation 0 0 -1 0
        children [
          Solid {
            translation 0 0.02 0.04
            rotation 1 0 0 1.5708
            children [
              HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 0 1
                }
                device [
                  RotationalMotor {
                    name "motor_pit"
                  }
                ]
                endPoint Solid {
                  translation 0 0 0.02
                  rotation 0 1 0 1.5708
                  children [
                    Solid {
                      translation 0 0 0.66
                      children [
                        Shape {
                          appearance Appearance {
                            material Material {
                              diffuseColor 0 1 0
                              transparency 0.3
                            }
                          }
                          geometry Cylinder {
                            height 1.3
                            radius 0.001
                          }
                        }
                      ]
                    }
                    Shape {
                      appearance Appearance {
                        material Material {
                          diffuseColor 0.192157 0.192157 0.192157
                        }
                      }
                      geometry Cylinder {
                        height 0.04
                        radius 0.005
                      }
                    }
                  ]
                }
              }
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.427451 0.427451 0.427451
                  }
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.02
                }
              }
            ]
          }
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.254902 0.254902 0.254902
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.02
            }
          }
        ]
      }
    }
    Shape {
      appearance Appearance {
        material Material {
        }
      }
      geometry Box {
        size 0.15 0.3 0.02
      }
    }
  ]
  name "target"
  controller "controller_target"
}
Robot {
  translation 0 -0.6 0.01
  rotation 0 1 0 0
  children [
    DistanceSensor {
      translation -0.05 0 0.02
      rotation 0 0 1 1.5708
      name "distance_sensor"
      type "laser"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
      }
      device [
        RotationalMotor {
          name "motor_yaw"
        }
      ]
      endPoint Solid {
        translation 0 0 0.02
        rotation 0 0 -1 0
        children [
          Solid {
            translation 0 0.02 0.04
            rotation 1 0 0 1.5708
            children [
              HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 0 1
                }
                device [
                  RotationalMotor {
                    name "motor_pit"
                  }
                ]
                endPoint Solid {
                  translation 0 0 0.02
                  rotation 0 1 0 1.5708
                  children [
                    Solid {
                      translation 0 0 0.99
                      children [
                        Shape {
                          appearance Appearance {
                            material Material {
                              diffuseColor 1 0 0
                              transparency 0.3
                            }
                          }
                          geometry Cylinder {
                            radius 0.001
                          }
                        }
                      ]
                    }
                    Shape {
                      appearance Appearance {
                        material Material {
                          diffuseColor 0.192157 0.192157 0.192157
                        }
                      }
                      geometry Cylinder {
                        height 0.04
                        radius 0.005
                      }
                    }
                  ]
                }
              }
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.427451 0.427451 0.427451
                  }
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.02
                }
              }
            ]
          }
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.254902 0.254902 0.254902
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.02
            }
          }
        ]
      }
    }
    Shape {
      appearance Appearance {
        material Material {
        }
      }
      geometry Box {
        size 0.15 0.3 0.02
      }
    }
  ]
  name "track"
  controller "controller_track"
}
Transform {
  translation 1.0001 0 0.5
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
      }
      geometry Box {
        size 0.001 1 1
      }
    }
  ]
}
Transform {
  translation 1 0 0.5
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.890196 0.890196 0.890196
        }
      }
      geometry Box {
        size 0.001 0.5 0.5
      }
    }
  ]
}
